test/rules/src/https_everywhere_checker/rules.py

```diff
@@ -80,6 +80,12 @@ def __init__(self, url):
         """
         self.url = url
 
+    def __eq__(test1, test2):
+        return test1.__hash__() == test2.__hash__()
```

Comment:
Why can't you use `return test1.url == test2.url` here, and remove the `__hash__` method below?  This would also remove the internal string `__hash__` method, which is preferable.